gensim/models/ldaseqmodel.py

```diff
@@ -0,0 +1,1096 @@
+
+
+    def init_ldaseq_ss(self, topic_chain_variance, topic_obs_variance, alpha, init_suffstats):
+        """
+        Method to initialize State Space Language Model, topic wise.
+        """
+        self.alphas = alpha
+        for k in range(0, self.num_topics):
+            sstats = init_suffstats[:, k]
+            sslm.sslm_counts_init(self.topic_chains[k], topic_obs_variance, topic_chain_variance, sstats)
+
+            # initialize the below matrices only if running DIM
+            # ldaseq.topic_chains[k].w_phi_l = numpy.zeros((ldaseq.vocab_len, ldaseq.num_time_slices))
+            # ldaseq.topic_chains[k].w_phi_sum = numpy.zeros((ldaseq.vocab_len, ldaseq.num_time_slices))
+            # ldaseq.topic_chains[k].w_phi_sq = numpy.zeros((ldaseq.vocab_len, ldaseq.num_time_slices))
+
+
+    def fit_lda_seq(self, seq_corpus):
+        """
+        fit an lda sequence model:
+
+        for each time period
+            set up lda model with E[log p(w|z)] and \alpha
+            for each document
+                perform posterior inference
+                update sufficient statistics/likelihood
+
+        maximize topics
+
+       """
+        LDA_INFERENCE_MAX_ITER = 25
+        LDASQE_EM_THRESHOLD = 1e-4
+        LDA_SEQ_MIN_ITER = 6
+        LDA_SEQ_MAX_ITER = 20
+
+        num_topics = self.num_topics
+        vocab_len = self.vocab_len
+        data_len = seq_corpus.num_time_slices
+        corpus_len = seq_corpus.corpus_len
+
+        bound = 0
+        convergence = LDASQE_EM_THRESHOLD + 1
+        iter_ = 0
+
+        while iter_ < LDA_SEQ_MIN_ITER or ((convergence > LDASQE_EM_THRESHOLD) and iter_ <= LDA_SEQ_MAX_ITER):
+
+            print (" EM iter ", iter_)
+            print ("E Step")
+
+            old_bound = bound
+
+            # initiate sufficient statistics
+            topic_suffstats = []
+            for topic in range(0, num_topics):
+                topic_suffstats.append(numpy.resize(numpy.zeros(vocab_len * data_len), (vocab_len, data_len)))
+
+            # set up variables
+            gammas = numpy.resize(numpy.zeros(corpus_len * num_topics), (corpus_len, num_topics))
+            lhoods = numpy.resize(numpy.zeros(corpus_len * num_topics + 1), (corpus_len, num_topics + 1))
+            # compute the likelihood of a sequential corpus under an LDA
+            # seq model and find the evidence lower bound. This is the E - Step
+            bound, gammas = self.lda_seq_infer(seq_corpus, topic_suffstats, gammas, lhoods, iter_)
+            self.gammas = gammas
+
+            print ("M Step")
+
+            # fit the variational distribution. This is the M - Step
+            topic_bound = self.fit_lda_seq_topics(topic_suffstats)
+            bound += topic_bound
+
+            if ((bound - old_bound) < 0):
+                if LDA_INFERENCE_MAX_ITER < 10:
+                    LDA_INFERENCE_MAX_ITER *= 2
+                print ("Bound went down, increasing iterations to", LDA_INFERENCE_MAX_ITER)
```

Comment:
All these `print` statements will go away, right?

@bhargavvader it is a better practice to use logging from the start. Just use DEBUG level if you want the messages to go away in "normal" mode (logging threshold set at INFO+).